app/scripts/controllers/alert.js

```diff
@@ -0,0 +1,53 @@
+import ObservableStore from 'obs-store'
+
+/**
+ * @typedef {Object} AlertControllerInitState
+ * @property {Object} alertEnabledness - A map of any alerts that were suppressed keyed by alert ID, where the value
+ *   is the timestamp of when the user suppressed the alert.
+ */
+
+/**
+ * @typedef {Object} AlertControllerOptions
+ * @property {AlertControllerInitState} initState - The initial controller state
+ */
+
+export const ALERT_TYPES = {
+  unconnectedAccount: 'unconnectedAccount',
+}
+
+const defaultState = {
+  alertEnabledness: Object.keys(ALERT_TYPES)
+    .reduce(
+      (alertEnabledness, alertType) => {
+        alertEnabledness[alertType] = true
+        return alertEnabledness
+      },
+      {}
+    ),
```

Comment:
The mutation of the accumulator here, while valid, feels weird. 😅 Maybe something like this:

```suggestion
    .reduce((alertEnabledness, alertType) => ({
      ...alertEnabledness,
      [alertType]: true,
    }), {}),
```

Reply:
Huh, really? I mutated it intentionally. Why bother construct a new object each time?

Reply:
> Why bother construct a new object each time?

I won't disagree that mutation is faster, but the reduce construct is/feels like a functional construct that fits with a more "immutable" pattern. It's an unimportant nitpick.

Reply:
Sure, fair enough. I still think of it as a"functional construct" because the accumulator is "internal" to this reduce step, so it's still effectively presenting an immutable interface. 🤷‍♂️